### DISCUSSION TITLE
Signed urls protocol omission

### DISCUSSION THREAD

**Discussion by Lukasz93P** [laravel/framework#39061] [Ideas]
Hi,
I think that configuration value which would make signed url signature omit protocol would be great, currently it causes problems on environments with load balancers and SSL offloading, because in that config app receives all requests with http protocol but generates url with https protocol when forcing SSL is turned on (this url will be shown to the users browsers so they have to be https).
Currently I managed to make my app in configuration described above deal with signed urls but it require ugly overwriting whole `\Illuminate\Routing\UrlGenerator::signedRoute`, it's almost copy paste with some small changes, also registration of that custom implementation is quite cumbersome.

**Comment by lukebouch** [comment]
I just ran into this issue with an app I'm working on that is using a Cloudflare Argo tunnel. We need `https://` urls in the front end but all the traffic that comes in to the application is `http://` because it's coming from the tunnel.

**Reply by lukebouch** [reply]
We ended up finding a work-around where we have a middleware that replaces `http://` with `https://` in the response on the way out; but that is not ideal.
